### DISCUSSION TITLE
Replace `black`, `flake8`, `isort` with `ruff`

### DISCUSSION THREAD

**Discussion by c0rydoras** [qtile/qtile#4980] [General]
# [Ruff](https://docs.astral.sh/ruff/)

I propose replacing `black`, `flake8` and `isort` with [ruff](https://docs.astral.sh/ruff/)

Why?
- its one tool and not 3
- ruff is faster (in formatting, linting)
- has a lot more linting rules than flake8 (https://docs.astral.sh/ruff/rules/), s
- its [formatter](https://docs.astral.sh/ruff/formatter/) is a drop in replacement for black

How?
- configure ruff to do the same (or more) as black, isort and flake8 do right now
  - [isort](https://docs.astral.sh/ruff/rules/#isort-i) 
  - [flake8-logging-format](https://docs.astral.sh/ruff/rules/#flake8-logging-format-g)
  - [pep8-naming](https://docs.astral.sh/ruff/rules/#pep8-naming-n)
- adjust [pre-commit](https://docs.astral.sh/ruff/integrations/#pre-commit)
- delete `black`, `isort` and `flake8` configuration from `pyproject.toml`

---

PR: #4981

**Comment by elParaguayo** [comment]
We talked about this a while ago. Don't think there were any significant objections other than we weren't sure if it was mature enough at that point.

**Comment by c0rydoras** [comment]
I'd be happy to contribute this, if there is a chance of it getting merged :)

**Comment by c0rydoras** [comment]
Also I kinda forgot the most important thing, ruffs linter can fix things it finds
